crates/rpc/src/v04/method/trace_transaction.rs

```diff
@@ -0,0 +1,113 @@
+use pathfinder_common::{BlockId, TransactionHash};
+use pathfinder_executor::{CallError, Transaction};
+use serde::{Deserialize, Serialize};
+use tokio::task::JoinError;
+
+use crate::{compose_executor_transaction, context::RpcContext, executor::ExecutionStateError};
+
+use super::simulate_transactions::dto::TransactionTrace;
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct TraceTransactionInput {
+    transaction_hash: TransactionHash,
+}
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+pub struct TraceTransactionOutput(TransactionTrace);
+
+crate::error::generate_rpc_error_subset!(
+    TraceTransactionError: InvalidTxnHash,
+    NoTraceAvailable
+);
+
+impl From<ExecutionStateError> for TraceTransactionError {
+    fn from(value: ExecutionStateError) -> Self {
+        match value {
+            ExecutionStateError::BlockNotFound => Self::NoTraceAvailable,
+            ExecutionStateError::Internal(e) => Self::Internal(e),
+        }
+    }
+}
+
+impl From<CallError> for TraceTransactionError {
+    fn from(value: CallError) -> Self {
+        match value {
+            CallError::ContractNotFound | CallError::InvalidMessageSelector => {
+                Self::Internal(anyhow::anyhow!("Failed to trace the block's transactions"))
+            }
+            CallError::Reverted(e) => Self::Internal(anyhow::anyhow!("Transaction reverted: {e}")),
```

Comment:
`trace_one()` doesn't seem to return `CallError::Reverted`. What is the expected output for reverted transactions in trace?

The spec doesn't mention that the call should fail entirely for reverted transactions but it seems that `to_trace()` does not properly handle `revert_error`. (In fact it seems that our `TransactionTrace` type can't even properly model the spec.)

It's also strange that only invoke transactions can be reverted according to the spec: we should probably clarify this with Starkware.

Reply:
That's not really what we should be doing. Instead, we should be returning proper traces for reverted transactions.